Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -93,5 +93,42 @@ TEST_F(SerializerTest, EncodesNullProtoToBytes) {
   ExpectRoundTrip(proto, bytes, FieldValue::Type::Null);
 }
 
+TEST_F(SerializerTest, EncodesBoolModelToProto) {
+  for (bool test : {true, false}) {
+    FieldValue model = FieldValue::BooleanValue(test);
+    Serializer::TypedValue proto{FieldValue::Type::Boolean,
+                                 google_firestore_v1beta1_Value_init_default};
```

Comment:
Continuing the discussion from the previous PR: why does `TypedValue` need to be POD?

Reply:
In general, I think it's preferable to have higher-level code, unless there is a specific reason not to. So in this case, hiding the `..._init_default` thing in a constructor, or perhaps a helper function, would avoid having to specify that non-intuitive initializer each time a `TypedValue` is created - it's repetitive and error-prone.

Reply:
Discussed offline. Summary:
* all nanopbs are init'd in this manner.
* _init_default is from the generated code.
* existing (POD) wrapper is similar in spirit to existing nanopb structs
* wrapping might still make sense (safer) even if it does end up being applied to all nanopb objects

Decision: Defer for now. (But others are welcome to chime in.)